apm-agent-core/src/main/java/co/elastic/apm/agent/logging/Log4j2ConfigurationFactory.java

```diff
@@ -121,7 +121,8 @@ public Configuration getConfiguration(LoggerContext loggerContext, Configuration
     Configuration getConfiguration() {
         ConfigurationBuilder<BuiltConfiguration> builder = newConfigurationBuilder();
         builder.setStatusLevel(Level.ERROR)
-            .setConfigurationName("ElasticAPM");
+            .setConfigurationName("ElasticAPM")
+            .setShutdownHook("disable");
```

Comment:
I had to change this so logging is still initialized during the shutdown hook.

@eyalkoren You did a lot in this area recently. Would you please have a look.

Reply:
Another option is to use `loggerContext.addShutdownListener()` within the other (API) `getConfiguration` overrides, but maybe using our own shutdown hook is preferable. Let me get back to you on that

Reply:
OK, let's disable the log4j shutdown hook and replace it with the existing mechanism we already have within the agent for this purpose: make `Log4jLoggerFactoryBridge` implement `co.elastic.apm.agent.context.LifecycleListener` and use the `stop` API to do both the logging and the shutdown

Reply:
A downside of the `LifecycleListener#stop` approach is that `Tracer switched to STOPPED state` won't get logged anymore.

Reply:
Aso `Tracer stop stack trace: ` won't be logged anymore

Reply:
Good point and even if we fix that by lifting these loggings a bit higher, there would be a race with other listeners, like `MatcherTimerLifecycleListener`.
So let's revert that and call the shutdown from the end of `co.elastic.apm.agent.impl.ElasticApmTracer#stop` instead. Sorry about that...

Reply:
Done